### PR TITLE
Adding Rect.expand and Rect.longestSide

### PR DESCRIPTION
**Pull request by gspencergoog**
Adding two functions to make the Rect API more balanced.

Rect.expand will return the bounding box of the union of the current rect and the supplied rect.

Rect.longestSide will return the max of the width and the height.

### CODE REVIEW

File: testing/dart/rect_test.dart [hunk 1]
```diff
@@ -8,30 +8,56 @@ import 'package:test/test.dart';
...
     expect(r3.left, equals(50.0));
     expect(r3.top, equals(50.0));
     expect(r3.right, equals(100.0));
     expect(r3.bottom, equals(100.0));
-    Rect r4 = r2.intersect(r1);
+    final Rect r4 = r2.intersect(r1);
+    expect(r4, equals(r3));
+  });
+
+  test("rect expand overlapping rects", () {
```
Comment (Hixie) [inline testing/dart/rect_test.dart]: trivial nit: we prefer `'` for consistency

Comment (Hixie) [reply testing/dart/rect_test.dart]: i guess the whole file uses " for some reason. Feel free to ignore.

File: lib/ui/geometry.dart [hunk 2]
```diff
@@ -653,6 +653,17 @@ class Rect {
     );
   }
 
+  /// Returns a new rectangle which is the bounding box containing this
+  /// rectangle and the given rectangle.
+  Rect expand(Rect other) {
```
Comment (Hixie) [inline lib/ui/geometry.dart]: Can you elaborate on the way this would be used? I'm trying to figure out if `expand` is the intuitive term in real uses.

Comment (TyOverby) [reply lib/ui/geometry.dart]: `expandToInclude`?

**Comment by gspencergoog** [comment]
Also added longestSide to Size and RRect, and fixed some lint warnings.

File: lib/ui/geometry.dart (hunk 2 above)
Comment (Hixie) [reply lib/ui/geometry.dart]: Another option would be a constructor that takes a list of Rects.

`new Rect.boundingBox(<Rect>[...])`

Comment (Hixie) [reply lib/ui/geometry.dart]: if it's to be a method, it could be `expandFor`

**Comment by Hixie** [comment]
LGTM

**Pull request merged**